nixos/modules/services/databases/neo4j.nix

```diff
@@ -36,81 +89,414 @@ let
     dbms.jvm.additional=-XX:+TrustFinalNonStaticFields
     dbms.jvm.additional=-XX:+DisableExplicitGC
     dbms.jvm.additional=-Djdk.tls.ephemeralDHKeySize=2048
-
+    dbms.jvm.additional=-Djdk.tls.rejectClientInitiatedRenegotiation=true
     dbms.jvm.additional=-Dunsupported.dbms.udc.source=tarball
+
+    # Usage Data Collector
+    dbms.udc.enabled=${boolToString cfg.udc.enable}
+
+    # Extra Configuration
+    ${cfg.extraServerConfig}
```

Comment:
I think this module is a good candidate to convert to a semantic config thing, similar to how I did it in #41467. The idea is to represent the full config as a Nix value, so something like

```nix
services.neo4j.config = {
  dbms.jvm.additional = [
    "-XX:+UseG1GC"
    ...
  ];
  dbms.shell.enabled = true
}
```

Then use a function to convert this to a neo4j readable config file. The advantages are:
- Users can easily override whatever they want instead of being bounded to the given config file
- The config gets a free syntax check
- More modular via NixOS module system, merging of values
- This module gets cleaner, less stringy

I'll gladly help with getting this to work

It's not necessary, but I'm very much in favor of doing this.

Reply:
> I think this module is a good candidate to convert to a semantic config thing

For the entire module or just the `extraServerConfig` option?

Reply:
The entire config file, instead of using giant strings to declare the config, you'd use nix values. See [this](https://github.com/Infinisil/nixpkgs/blob/c9ee4f9136dafb767837505edb7491d56c1c0e65/nixos/modules/services/networking/znc/opts.nix#L227-L253) as an example of how assigning default values would look like.